src/commonMain/kotlin/uk/gov/hmrc/calculator/utils/validation/PensionValidator.kt

```diff
@@ -17,14 +17,47 @@ package uk.gov.hmrc.calculator.utils.validation
 
 import uk.gov.hmrc.calculator.model.TaxYear
 import uk.gov.hmrc.calculator.model.pension.PensionAllowances.getPensionAllowances
+import uk.gov.hmrc.calculator.utils.validation.HoursDaysValidator.isTwoDecimalPlacesOrFewer
+import kotlin.jvm.JvmSynthetic
 
 object PensionValidator {
 
-    fun isValidYearlyPension(yearlyWage: Double, yearlyPension: Double): Boolean {
+    fun isValidYearlyPension(yearlyPension: Double, wage: Double, taxYear: TaxYear): MutableList<PensionError> {
+        val listOfError = mutableListOf<PensionError>()
+
+        if (!yearlyPension.isTwoDecimalPlacesOrFewer() || !isPensionValidFormat(yearlyPension)) {
+            listOfError.add(PensionError.INVALID_FORMAT)
+        }
+        if (isPensionBelowZero(yearlyPension)) listOfError.add(PensionError.BELOW_ZERO)
+        if (!isPensionLowerThenWage(yearlyPension, wage)) listOfError.add(PensionError.ABOVE_HUNDRED_PERCENT)
+        if (isAboveAnnualAllowance(yearlyPension, taxYear)) listOfError.add(PensionError.ABOVE_ANNUAL_ALLOWANCE)
+
+        listOfError.sortBy { it.priority }
+        return listOfError
+    }
+
+    private fun isPensionValidFormat(yearlyPension: Double): Boolean {
+        return "([0-9])+(\\.\\d{1,2})".toRegex().matches(yearlyPension.toString())
+    }
+
+    private fun isPensionBelowZero(yearlyPension: Double): Boolean {
+        return yearlyPension <= 0.0
+    }
+
+    @JvmSynthetic
+    internal fun isPensionLowerThenWage(yearlyPension: Double, yearlyWage: Double): Boolean {
         return yearlyPension <= yearlyWage
     }
 
-    fun isAboveAnnualAllowance(yearlyPension: Double, taxYear: TaxYear): Boolean {
+    @JvmSynthetic
+    internal fun isAboveAnnualAllowance(yearlyPension: Double, taxYear: TaxYear): Boolean {
```

Comment:
For consistency with the naming of the other functions, and clarity of which allowance.. 

```suggestion
    internal fun isPensionAboveAnnualAllowance(yearlyPension: Double, taxYear: TaxYear): Boolean {
```